eqcorrscan/scripts/correlation_speeds.py

```diff
@@ -0,0 +1,176 @@
+        for i in mem:
+            sline = i.split()
+            if str(sline[0]) in ('MemFree:', 'Buffers:', 'Cached:'):
+                free_memory += int(sline[1])
+    return free_memory
+
+
+def generate_dataset(n_templates, n_stations, n_channels, data_len,
+                     template_len, sampling_rate):
+    """
+
+    :param n_templates:
+    :param n_stations:
+    :param n_channels:
+    :param data_len:
+    :param template_len:
+    :param sampling_rate:
+    :return:
+    """
+    print("=" * 80)
+    print("Generating dataset".center(80))
+    print("=" * 80)
+    dataset = {'data': Stream(), 'templates': []}
+    for i in range(n_stations * n_channels):
+        dataset['data'] += Trace(np.random.randn(
+            int(data_len * sampling_rate)))
+    for t in range(n_templates):
+        template = Stream()
+        for i in range(n_stations * n_channels):
+            template += Trace(np.random.randn(
+                int(template_len * sampling_rate)))
+        dataset['templates'].append(template)
+    return dataset
+
+
+def run_correlation(func, threads, dataset, loops=3):
+    """
+    Run the given correlation function and profile time and memory usage.
+
+    :param func:
+    :param threads:
+    :param dataset:
+    :param loops:
+    :return:
+    """
+    max_mem = []
+    timings = []
+    for loop in range(loops):
+        tic = time.time()
+        max_mem = memory_usage((func, (dataset['templates'], dataset['data']),
+                                {'cores': threads}), max_usage=True,
+                               include_children=True, multiprocess=True)
+        toc = time.time()
+        timings.append(toc - tic)
+    return np.mean(max_mem[0]), np.mean(timings)
+
+
+def plot_profiles(times, memory_use):
+    """
+
+    :param times:
+    :param memory_use:
+    :return:
+    """
+    plt.xkcd()
+    markers = itertools.cycle((',', '+', '.', 'o', '*'))
+    for key in times.keys():
+        plt.scatter(memory_use[key], times[key], s=10**2, marker=next(markers),
+                    label=key)
+    plt.xlabel('Memory use (MB)')
+    plt.ylabel('Time (s)')
+    plt.legend()
+    plt.show()
+
+
+def run_profiling(n_templates, n_stations, n_channels, data_len,
+                  template_len, sampling_rate, loops=3):
+    """
+
+    :param n_templates:
+    :param n_stations:
+    :param n_channels:
+    :param data_len:
+    :param template_len:
+    :param sampling_rate:
+    :return:
+    """
+    print("Testing correlation functions in %s" % correlate.__file__)
+    MAXTHREADS = cpu_count()
+    dataset = generate_dataset(
+        n_templates=n_templates, n_stations=n_stations, n_channels=n_channels,
+        data_len=data_len, template_len=template_len,
+        sampling_rate=sampling_rate)
+    times = {}
+    memory_use = {}
+    # Limit the memory
+    memory_limit()
+    for corr_func in XCOR_FUNCS.keys():
+        if corr_func == 'default':
+            continue
+        print("=" * 80)
```

Comment:
Consider creating a small helper function to do this kind of printing, will cut back on repeated code.

Reply:
:+1: